### DISCUSSION TITLE
Assets not working properly when webpack is being used in a Module as well as consumer end

### DISCUSSION THREAD

**Discussion by abhishek4794** [webpack/webpack#17321] [Q&A]
# Bug report

**What is the current behavior?**

I have one component library that is using Webpack.
I am using an asset loader for svgs,

```
{
        test: /\.(png|jp(e*)g|svg|gif)$/,
        type: "asset/resource",
}
```

In React component, I am using svg like this 

`import Cart from './cart.svg';`

When I am compiling this, 
Webpack is compiling it in this form

```
// CONCATENATED MODULE: ./src/Button/cart.svg
const cart_namespaceObject = __webpack_require__.p + "ee4d89699e3e3de82036.svg";
```

Now, the issue is, When I am trying to use this component in another consumer react Project which is also using webpack,

`__webpack_require__.p + "ee4d89699e3e3de82036.svg"; `
is not working properly,

The consumer's web pack is not changing any code or path here, and it's trying to access svg in the consumer's static assets location,
As that svg only exists inside the component library, its throwing an error.

Anything can be done for this one?

**If the current behavior is a bug, please provide the steps to reproduce.**

I am not sure if this is a bug or I am missing something.
But steps are like this

1. Creating one component library in react using Webpack
2. Using SVG inside that library
3. When I am trying to access that SVG in some another React project which is also using webpack, I am getting 404 error for that SVG
4. Webpack is not copying assets from library to its own static folder and adjust path accordingly.

**What is the expected behavior?**

Webpack should copy static assets from node_module into consumers static path and adjust webpack require statements for those assets.

**Other relevant information:**
webpack version:  5.82.1
Node.js version: 18.6.0
Operating System: Mac Os
Additional tools:

**Comment by alexander-akait** [comment]
Unfortunately, this cannot be solved, you can use `new URL(...)` and disable https://webpack.js.org/configuration/module/#moduleparserjavascripturl so webpack will keep `new URL(...)` syntax to the next application bundling or you can just ignore such imports, there is not `import ... from "./file.svg";` in ESM, you can't use `svg` extensions in ESM right now
